drivers/shared/executor/executor.go

```diff
@@ -303,8 +306,30 @@ func (e *UniversalExecutor) Launch(command *ExecCommand) (*ProcessState, error)
 	e.childCmd.Env = e.commandCfg.Env
 
 	// Start the process
-	if err := e.childCmd.Start(); err != nil {
-		return nil, fmt.Errorf("failed to start command path=%q --- args=%q: %v", path, e.childCmd.Args, err)
+	if command.NetworkIsolation != nil && command.NetworkIsolation.Path != "" {
+		// Lock to the thread we're changing the network namespace of
+		runtime.LockOSThread()
```

Comment:
Is this necessary?  Seeing `ns` code, I see that the `Do` function is called in a different goroutine and that goroutine is already protected with `runtime.LockOSThread()` in https://github.com/containernetworking/plugins/blob/fe60fcddb897079746ec1523fd1837ab05b1e689/pkg/ns/ns_linux.go#L197-L201 ; and I don't see runtime call in the CNI example in https://github.com/hwchiu/CNI_Tutorial_2018/blob/ea43c401127955a1e8974cfd03829509c17f214a/tutorial/step3/example.go .

If needed, let's comment the need and also point to why we shouldn't be calling `runtime.UnlockOSThread()`.

I

Reply:
You are correct, I think I talked about this with someone else too. Calling `runtime.LockOSThread()` is sort of a muscle memory I've developed from working with network namespaces before this `ns.Do` block existed.